File: src/backend/Domain/Models/Movie.cs
namespace Domain.Models;

public class Movie
{
    public Guid Id { get; set; }
    public string Title { get; set; }
    public string Description { get; set; }
    public int Year { get; set; }
    public int DurationAtMinutes { get; set; }
    public string AgeRating { get; set; }
    public int RatingCount { get; set; }
    public double RatingSum { get; set; }
    public string PosterUrl { get; set; }
    
    public List<Person> Directors { get; set; } = new();
    public List<Person> Writers { get; set; } = new();
    public List<Genre> Genres { get; set; } = new();
    public List<Photo>? Photos { get; set; }
    public List<MovieActor> Actors { get; set; } = new();
}

File: src/frontend/src/models/movie.ts
import type {MovieDiscussionTopicResponse} from "./topic.ts";

export interface PersonResponse {
  id: string;
  firstName: string;
  lastName: string;
}

export interface ActorCardResponse {
  id: string;
  firstName: string;
  lastName: string;
  photoUrl: string;
  characterName: string;
}

export interface MovieData {
  id: string;
  title: string;
  description: string;
  year: number;
  durationAtMinutes: number;
  ageRating: string;
  userRating: number;
  ratingCount: number;
  posterUrl: string;
  directors: PersonResponse[];
  writers: PersonResponse[];
  movieActors: ActorCardResponse[];
  topics:MovieDiscussionTopicResponse[];
  genres: string[];
  photos: string[];
}

export interface MovieSearchResponse {
  id: string;
  title: string;
  year: number;
  userRating: number;
  actors: ActorSearchResponse[];
}

export interface ActorSearchResponse {
  id: string;
  firstName: string;
  lastName: string;
  photoUrl: string;
}

export interface MovieResponse {
  movie: MovieData;
  isInWatchList: boolean;
  ownRating?: number;
}

